synapse/lib/storm.py

```diff
@@ -1536,7 +1536,13 @@ def _stormOperTree(self, query, oper):
         core = self.getStormCore()
 
         # Prevent infinite pivots
-        recurlim, _ = core.getTypeNorm('int', opts.get('recurlim', 20))
+        try:
+            recurlim, _ = core.getTypeNorm('int', opts.get('recurlim', 20))
+        except s_common.BadTypeValu as e:
+            raise s_common.BadOperArg(oper='tree', name='recurlim', mesg=e.errinfo.get('mesg'))
```

Comment:
I think that it makes sense to get consistent exceptions for incorrect use of opers